builder/triton/ssh.go

```diff
@@ -0,0 +1,36 @@
+package triton
+
+import (
+	"fmt"
+
+	"github.com/mitchellh/multistep"
+	"golang.org/x/crypto/ssh"
+)
+
+func commHost(state multistep.StateBag) (string, error) {
+	driver := state.Get("driver").(Driver)
+	machineID := state.Get("machine").(string)
+
+	machine, err := driver.GetMachine(machineID)
+	if err != nil {
+		return "", err
+	}
+
+	return machine, nil
+}
+
+func sshConfig(state multistep.StateBag) (*ssh.ClientConfig, error) {
+	config := state.Get("config").(Config)
+
+	signer, err := ssh.ParsePrivateKey([]byte(config.Comm.SSHPrivateKey))
```

Comment:
I encountered the following error:

```
[DEBUG] Error getting SSH config: Error parsing SSH private key: ssh: no key found
```

`config.Comm.SSHPrivateKey` holds the path to the file containing the private key. Not the raw private key it self. The function `ParseRawPrivateKey` which is later called expects the raw private key in a byte[].

The following change fixes the problem:
```
privateKeyBytes, err := ioutil.ReadFile(config.Comm.SSHPrivateKey)
if err != nil {
	return nil, fmt.Errorf("Unable to read private key from file '%s': %s", config.Comm.SSHPrivateKey, err)
}

signer, err := ssh.ParsePrivateKey(privateKeyBytes)
```